libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/gutenberg/GutenbergEditorFragment.java

```diff
@@ -1189,8 +1192,14 @@ public void onMediaUploadProgress(final String localMediaId, final float progres
     }
 
     @Override
-    public void onMediaUploadFailed(final String localMediaId) {
-        getGutenbergContainerFragment().mediaFileUploadFailed(Integer.valueOf(localMediaId));
+    public void onMediaUploadFailed(final String localMediaId, final String errorMessage) {
+        if (errorMessage.equals(FILE_TYPE_NOT_PERMITTED_ERROR)) {
+            getGutenbergContainerFragment().mediaFileUploadFailed(Integer.valueOf(localMediaId),
+                    getString(R.string.error_file_type_not_supported));
+        } else {
+            getGutenbergContainerFragment().mediaFileUploadFailed(Integer.valueOf(localMediaId), null);
+        }
```

Comment:
I need some feedback on the direction I decided to take here. My thoughts were as follows. Currently, when the we try to upload an audio file on a free site we get the error "Sorry, this file type is not permitted for security reasons." So my idea was that once this occurs then we should tell the user that the file type is not supported. 
However, when I did a search online I realized that [this error occurs for scenarios](https://kinsta.com/knowledgebase/sorry-this-file-type-is-not-permitted-for-security-reasons/) other than the site being free or not. A primary example ( contained in the mentioned link) is when a mime type is not supported. So is it okay to keep this behavior for now until we implemented the ability to detect which mime types are supported by the site's plan?

Reply:
Could there be any more clues inside the error we receive that's the same between other types of sites for this kind of error? Like HTTP status code or header fields? Would that be possible to pass those in here somehow?

Reply:
Good question. I am going to investigate if there exists anything specific for this kind of error.

Reply:
I agree with Ceyhun that it might be better to check for another way to detect the error other than matching the error.message against that specific string. If after more digging this is the only way we can find, some comments explaining the Strings origin would be helpful.